### PR TITLE
Update 'Arrow Function' wording for clarity.

### PR DESCRIPTION
**Pull request by merriam**
(minor doc change).  Saw a surprising number of issues and stack overflow questions about using arrow functions, so updated the documentation to be explicit about the lexical binding of 'this'.

### CODE REVIEW

**Comment by ScottFreeCode** [comment]
This is a nice simplification, and addressing a surprisingly common issue; thanks! I'm almost inclined to dumb it down/spell it out even further, something like...
```txt
Passing [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) ("lambdas") to Mocha is discouraged.  Arrow functions do not get their own `this` reference and therefore cannot access the Mocha context object.  For example, the following code will fail:
```

If we really want to talk about what arrow functions do get as `this`, we could add "(instead inheriting the \`this\` object of the function in which they are written)" after "their own \`this\` reference", but I'm inclined to say people can follow the link if they want to know more -- the important part we should be focusing on is, you can't use them and get the `this` object that Mocha passes to the function when calling it.

(I do worry in any case that "lexical binding" being meaningless to too many programmers might be just as much an obstacle to understanding as the current complexity of the sentence. I wish that it was not so, that everyone were familiar with the terminology, but...)

**Comment by merriam** [comment]
I'm fine with either change; both wordings solve the problem.  Please pick one, merge it, and we can move on to the next confusing bit. :)   Have a great day. -- Charles

**Comment by boneskull** [comment]
@merriam thanks.  I'm merging this via https://github.com/mochajs/mocha/pull/3123
